rs/tests/networking/canister_http_correctness_tests/canister_http_correctness_test.rs

```diff
@@ -0,0 +1,776 @@
+    assert_eq!(
+        response.body.as_str(),
+        "transform_context",
+        "Transform function did not set the body to the provided context."
+    );
+    assert_eq!(response.status, 202);
+}
+
+fn test_non_existent_transform_function(env: TestEnv) {
+    let request = non_existent_transform_function(&env);
+
+    let handlers = Handlers::new(&env);
+    let response = block_on(submit_outcall(&handlers, request));
+    assert_matches!(
+        response,
+        Err(RejectResponse {
+            reject_code: RejectCode::CanisterError,
+            ..
+        })
+    );
+}
+
+fn test_composite_transform_function_is_executed(env: TestEnv) {
+    let request = composite_transform_function_is_executed(&env);
+
+    let handlers = Handlers::new(&env);
+    let response = block_on(submit_outcall(&handlers, request));
+    let response = response.expect("Http call should succeed");
+    assert_eq!(response.headers.len(), 2, "Headers: {:?}", response.headers);
+    assert_eq!(response.headers[0].0, "hello");
+    assert_eq!(response.headers[0].1, "bonjour");
+    assert_eq!(response.headers[1].0, "caller");
+    assert_eq!(response.headers[1].1, "aaaaa-aa");
+}
+
+fn test_no_cycles_attached(env: TestEnv) {
+    let request = no_cycles_attached(&env);
+
+    let handlers = Handlers::new(&env);
+    let response = block_on(submit_outcall(&handlers, request));
+    assert_matches!(
+        response,
+        Err(RejectResponse {
+            reject_code: RejectCode::CanisterReject,
+            ..
+        })
+    );
+}
+
+fn test_max_possible_request_size(env: TestEnv) {
+    let request = max_possible_request_size(&env);
+
+    let handlers = Handlers::new(&env);
+    let response = block_on(submit_outcall(&handlers, request));
+    assert_matches!(response, Ok(r) if r.status==200);
+}
+
+fn test_max_possible_request_size_exceeded(env: TestEnv) {
+    let request = max_possible_request_size_exceeded(&env);
+
+    let handlers = Handlers::new(&env);
+    let response = block_on(submit_outcall(&handlers, request));
+    assert_matches!(
+        response,
+        Err(RejectResponse {
+            reject_code: RejectCode::CanisterReject,
+            ..
+        })
+    );
+}
+
+fn test_size_2mb_response_insufficient_cycle_for_rejection_path(env: TestEnv) {
+    let request = size_2mb_response_insufficient_cycles_for_rejection_path(&env);
+
+    let handlers = Handlers::new(&env);
+    let response = block_on(async move { submit_outcall(&handlers, request).await });
+    assert_matches!(
+        response,
+        Err(RejectResponse {
+            reject_code: RejectCode::CanisterReject,
+            ..
+        })
+    );
+}
+
+fn test_size_4096_max_response_cycle_case_1(env: TestEnv) {
+    let request = size_4096_max_response_cycle_case_1(&env);
+
+    let handlers = Handlers::new(&env);
+    let response = block_on(submit_outcall(&handlers, request));
+    assert_matches!(response, Ok(r) if r.status==200);
```

Comment:
Maybe this looks nicer? Up to you
```suggestion
    assert_matches!(response, Ok(RemoteHttpResponse {status: 200, ... }));
```